docs/docs/api_docs/dataframe_lib.md

```diff
@@ -0,0 +1,116 @@
+# Bodo Pandas API (bodo.pandas)
+
+The bodo.pandas library is designed to scale Pandas workflows with just a one-line change — simply replace:
+
+``` py
+import pandas as pd
+```
+
+with
+
+``` py
+import bodo.pandas as pd
+```
+
+and your existing code can immediately take advantage of high-performance, scalable execution.
+
+Key features include:
+
+- __Full Pandas API compatibility__ with a transparent fallback mechanism to native Pandas, ensuring that your workflows continue uninterrupted even if a feature is not yet supported.
+
+- __Lazy evaluation of operations__, allowing powerful optimizations like filter pushdown and column pruning behind the scenes.
+
+- __Scalable MPI-based execution__, leveraging HPC (High-Performance Computing) techniques for efficient parallelism; whether you're working on a laptop or running jobs across a large cloud cluster.
+
+- __Vectorized execution__ with streaming and spill-to-disk capabilities, making it possible to process datasets larger than memory reliably.
+
+!!! warning
+    `bodo.pandas` is under active development and is currently considered experimental. Some features and APIs may not yet be fully supported. We welcome your feedback—please join our community Slack or open an issue on GitHub if you encounter any problems!
+
+## Lazy Evaluation and Fallback to Pandas
+
+`bodo.pandas` operates with lazy evaluation, meaning operations are recorded into a query plan rather than executed immediately. Execution is automatically triggered only when results are actually needed, such as when displaying the first 5 rows in a DataFrame with `print(df.head(5))`.
```

Comment:
```suggestion
Bodo DataFrame library operates with lazy evaluation to allow query optimization, meaning operations are recorded into a query plan rather than executed immediately. Execution is automatically triggered only when results are actually needed, such as when displaying the first 5 rows in a DataFrame with `print(df.head(5))`.
```